backend/routes/league.go

```diff
@@ -1,11 +1,102 @@
 package routes
 
-// s := r.PathPrefix("/league").Subrouter()
-// // "/products/"
-// s.HandleFunc("/postleague", AddLeague)
-// // "/products/{key}/"
-// s.HandleFunc("/getLeagueById/{ID}", GetLeagueById)
-// // "/products/{key}/details"
-// s.HandleFunc("/deleteLeagueById/{ID}", DeleteLeague)
-// // "/league/deleteLeagueById/{ID}"
-// s.HandleFunc("/updateLeagueById/{ID}", UpdateLeague)
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/grommers00/civitas/backend/models"
+
+	"github.com/gorilla/mux"
+)
+
+// ConnectLeaguesSubrouter adds the different restFUL apis to the main router
+func ConnectLeaguesSubrouter(r *mux.Router) *mux.Router {
+	// "/leagues/"
+	f := r.PathPrefix("/leagues").Subrouter()
+
+	// "/leagues/"
+	f.HandleFunc("/", AddLeagues).Methods("POST")
+
+	// "/leagues/"
+	f.HandleFunc("/", GetAllLeagues).Methods("GET")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", GetLeaguesByID).Methods("GET")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", DeleteLeaguesByID).Methods("DELETE")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", UpdateLeaguesByID).Methods("PATCH")
+
+	return f
+}
+
+// AddLeagues will add a new article the list of articles
+func AddLeagues(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("leagues Added!\n"))
+}
+
+// GetAllLeagues gets all the leagues articles
+func GetAllLeagues(w http.ResponseWriter, r *http.Request) {
+	// Reading from json file
```

Comment:
So I'm aware that this is a temp solution while we await the database work, but I'd suggest (based on the same logic being repeated throughout the file and also the news routes, so let's separate this to a dedicated function for reading the json file.

Reply:
sounds good, I agree that would be a much better way to implement, I'll make that change later today.